components/dashboard/src/workspaces/WorkspaceEntry.tsx

```diff
@@ -111,18 +143,32 @@ export function WorkspaceEntry({ desc, model, isAdmin, stopWorkspace }: Props) {
             </Tooltip>
         </ItemField>
         <ItemFieldContextMenu menuEntries={menuEntries} />
-        {isModalVisible && <ConfirmationModal
+        {isDeleteModalVisible && <ConfirmationModal
             title="Delete Workspace"
             areYouSureText="Are you sure you want to delete this workspace?"
             children={{
                 name: ws.id,
                 description: ws.description,
             }}
             buttonText="Delete Workspace"
-            visible={isModalVisible}
-            onClose={() => setModalVisible(false)}
+            visible={isDeleteModalVisible}
+            onClose={() => setDeleteModalVisible(false)}
             onConfirm={() => model.deleteWorkspace(ws.id)}
         />}
+        <Modal visible={isRenameModalVisible} onClose={() => setRenameModalVisible(false)} onEnter={() => { updateWorkspaceDescription(); return true }}>
+            <h3 className="mb-4">Rename Workspace Description</h3>
+            <div className="border-t border-b border-gray-200 dark:border-gray-800 -mx-6 px-6 py-4 space-y-2">
+                <input className="w-full" type="text" defaultValue={workspaceDescription} ref={renameInputRef} />
+                <div className="mt-1">
+                    <p className="text-gray-500">You can use a friendlier name for this workspace description.</p>
+                    <p className="text-gray-500"> Workspace URLs and endpoints will remain the same.</p>
+                </div>
+            </div>
+            <div className="flex justify-end mt-6">
+                <button className="secondary" onClick={() => setRenameModalVisible(false)}>Cancel</button>
+                <button className="ml-2" type="submit" onClick={updateWorkspaceDescription}>Rename</button>
```

Comment:
**suggestion**: Using a more descriptive button label following the pattern used elsewhere sounds good. What do you think about this?

```suggestion
                <button className="ml-2" type="submit" onClick={updateWorkspaceDescription}>Update Description</button>
```

| Variables | Integrations | Workspaces |
|-|-|-|
| <img width="537" alt="Screenshot 2021-09-15 at 7 36 04 PM" src="https://user-images.githubusercontent.com/120486/133473977-657e0cca-37b2-4698-a86f-6a158ff496c6.png"> | <img width="537" alt="Screenshot 2021-09-15 at 7 36 29 PM" src="https://user-images.githubusercontent.com/120486/133474066-1a89edc4-e9f2-4d1a-ab29-5fa837f63447.png"> | <img width="539" alt="Screenshot 2021-09-15 at 7 36 42 PM" src="https://user-images.githubusercontent.com/120486/133474080-b1723566-0794-4e1e-ad61-ae0d03c2e2be.png"> |

Reply:
I committed the suggestion but somehow it still shows the old "Rename" button 🤔